qiita_db/metadata_template.py

```diff
@@ -1561,9 +1561,16 @@ def update(self, md_template):
                                % ', '.join(samples_diff))
         columns_diff = set(new_map.columns) - set(current_map.columns)
         if columns_diff:
-            raise QiitaDBError('The new sample template differs from what is '
-                               'stored in database by these columns names: %s'
-                               % ', '.join(columns_diff))
+            queue = 'add_cols_%d' % self._id
+            datatypes = dict(izip(new_map.columns, _get_datatypes(new_map)))
+            conn_handler.create_queue(queue)
+
+            for col in columns_diff:
+                self.add_category(col, new_map[col].to_dict(), datatypes[col])
```

Comment:
This is going to leverage a lot of connection openings/closing to the database, with the added danger of leveraging the db in an inconsistent state. You can either (a) duplicate the code of that function here so everything is executed in a single queue; or (b) modify the function signature so a conn_handler and a queue name can be passed, so it just gets added to the queue; or (c) do a private function to add things to a queue which is shared by both functions.

My vote is (c) as that pattern has been already done in other portions of the code (the data objects) and it is the one that maintains a consistency on the code and in the functions (i.e. each function only has a possible output w/o side effects depending on the input parameters)